HW1/P2/P2b.py

```diff
@@ -1,3 +1,42 @@
+from __future__ import division
 from P2 import *
+import pyspark
+import time
 
-# Your code here
+
+sc = pyspark.SparkContext(appName = "Spark1")
+
+baseList = np.array([ii for ii in range(0,2000)])
+np.random.seed(100)
+rList = np.random.uniform(0,1,2000)
+rPerm = np.array(sorted(range(len(rList)), key = lambda k:rList[k])) # get random permutation of 1,...,2000
+repNum = np.array([ii for ii in range(0,10)]*200) # get 200 copies of each hash key (1,...,10)
+rDictX = repNum[rPerm] # form hash map using random permutation of hash keys, same for y values below
+np.random.seed(200)
+rList = np.random.uniform(0,1,2000)
+rPerm = np.array(sorted(range(len(rList)), key = lambda k:rList[k]))
+repNum = np.array([ii for ii in range(0,10)]*200)
+rDictY = repNum[rPerm]
```

Comment:
When you call an identical chunk of code twice, it makes sense to define a separate function and then call that function